server/src/apub/comment.rs

```diff
@@ -372,3 +388,54 @@ impl ApubLikeableType for Comment {
     Ok(())
   }
 }
+
+struct MentionsAndAddresses {
+  addressed_ccs: Vec<String>,
+  inboxes: Vec<String>,
+  tags: Vec<Mention>,
+}
+
+fn collect_non_local_mentions_and_addresses(
+  conn: &PgConnection,
+  content: &str,
+  community: &Community,
+) -> Result<MentionsAndAddresses, Error> {
+  let mut addressed_ccs = vec![community.get_followers_url()];
+
+  // Add the mention tag
+  let mut tags = Vec::new();
+
+  // Get the inboxes for any mentions
+  let mentions = scrape_text_for_mentions(&content)
+    .into_iter()
+    // Filter only the non-local ones
+    .filter(|m| !m.is_local())
+    .collect::<Vec<MentionData>>();
+  let mut mention_inboxes = Vec::new();
+  for mention in &mentions {
+    // TODO should it be fetching it every time?
+    if let Ok(actor_id) = fetch_webfinger_url(mention) {
```

Comment:
You should only have to use webfinger if the user is not in the local db.

Reply:
But the problem is a mention doesn't know the actual `actor_id` (at least for other platforms) which is the whole reason for fetching the webfinger in the first place.

If I have `@user@instance.com`, I can only guess that their actor_id is `https://instance.com/u/user`.

Reply:
But you can just check the local database for a user with that name and domain? If that exists, you get the ID and otherwise you do the webfinger fetch.